sources.sh

```diff
@@ -219,6 +219,22 @@ jq <<<"$sources" --argjson pins "$externalPinsJson" '
 		# https://unix.stackexchange.com/a/738744/153467
 		reduce .[] as $a ([]; if IN(.[]; $a) then . else . += [$a] end)
 	;
+	def meld($a; $b):
+		# recursive merge of objects like "*", but also append lists (uniquely) instead of replace
+		# https://stackoverflow.com/a/53666584, but with lists unique-ified
+		if ($a | type) == "object" and ($b | type) == "object" then
+			reduce ([$a, $b] | add | keys_unsorted[]) as $k ({};
+				.[$k] = meld( $a[$k]; $b[$k])
+			)
+		elif ($a | type) == "array" and ($b | type) == "array" then
+			$a + $b
+			| unique_unsorted
+		elif $b == null then
+			$a
+		else
+			$b
+		end
```

Comment:
Even better, ditch `$a` and use `.` and everything simplifies:

```suggestion
	def meld($o):
		# recursive merge of objects like "*", but also append lists (uniquely) instead of replace
		# https://stackoverflow.com/a/53666584, but with lists unique-ified and less work being done
		if type == "object" and ($o | type) == "object" then
			reduce ($o | keys_unsorted[]) as $k (.;
				.[$k] |= meld($o[$k])
			)
		elif type == "array" and ($o | type) == "array" then
			. + $o
			| unique_unsorted
		elif $o == null then
			.
		else
			$o
		end
```

(untested)